### PR TITLE
[Foundation] Fix several styling errors with attachment forms

### PR DESCRIPTION
**Pull request by floriank**
Change the way form--field padding is attached.

This also includes an updated version for the attachment partials, which should be
merged in future versions, making it into a component.

Meets the second part of https://community.openproject.org/work_packages/18927

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://floriank-fix-18927-fix-attachment-alignment-issues.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://floriank-fix-18927-fix-attachment-alignment-issues.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by myabc** [comment]
Renaming: _Zurb_ is the company, _Foundation_ is the product :smile:

**Comment by floriank** [comment]
I know.

I debated with @NobodysNightmare which sounds better - I have been referring to it in the context of the epic :laughing: 

It's just a marker for me and @ulferts to give this a _premium review_<sup>TM</sup> as I want to finish the epic. could be called [NoOneCares] for all I care.

File: app/assets/stylesheets/content/_forms.sass [hunk 1]
```diff
@@ -217,15 +217,21 @@ fieldset.form--fieldset
...
-    padding-right: 1rem
+  // .grid-block:nth-last-of-type(n+2) > &,
+  // .form--row:nth-last-of-type(n+2) > &,
+  // .form--grouping-row > &:nth-last-of-type(n+2)
+  //   padding-right: 1rem
+
+  .grid-block > &:last-child,
+  .form--row > &:last-child,
+  .form--grouping-row > &:last-child
+    padding-right: 0
```
Comment (ulferts) [inline app/assets/stylesheets/content/_forms.sass:234]: @myabc I would like to get your feedback on this one as you created the rule initially. In any case @floriank, the commented out code is to be removed.

**Comment by ulferts** [comment]
@floriank reformulating the rule changed this:
image
into this:
image

**Comment by floriank** [comment]
Fuck. This cannot be that difficult to unify for all forms, cannit?

**Comment by ulferts** [comment]
https://community.openproject.org/work_packages/18927 also complains about the distance between the file selector and it's associated description field. @floriank, @myabc we could make the one field flex shrink and the other flex expand to fix that but it is somewhat of a fundamental decision, and I would like to have your opinons on that. If we are to do it it should be a modifier css class and find it's way into the styleguide.

**Comment by ulferts** [comment]
@floriank when tackling that nth-last-of-type beast again, please also take a look here: https://community.openproject.org/work_packages/18680 it seems to be related.

**Comment by ulferts** [comment]
I think part of the problem is fixed with #2725.

**Comment by floriank** [comment]
I will close this for now and have another look at it, once #2725 is merged.

**Pull request closed without merging**